sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/TestInputFileBuilder.java

```diff
@@ -52,24 +68,54 @@
   private int nonBlankLines;
   private int[] originalLineOffsets;
   private boolean publish = true;
+  private String contents;
 
+  /**
+   * Create a InputFile identified by the given module key and relative path.
+   * The module key will also be used as the module's base directory. 
+   */
   public TestInputFileBuilder(String moduleKey, String relativePath) {
     this(moduleKey, relativePath, batchId++);
   }
 
+  /**
+   * Create a InputFile with a given module key and module base directory. 
+   * The relative path is generated comparing the file path to the module base directory. 
+   * filePath must point to a file that is within the module base directory.
+   */
+  public TestInputFileBuilder(String moduleKey, File moduleBaseDir, File filePath) {
+    String relativePath = moduleBaseDir.toPath().relativize(filePath.toPath()).toString();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "relativePath" which hides the field declared at line 58. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck)